src/stream_management/mod_stream_management_stale_h.erl

```diff
@@ -0,0 +1,136 @@
+-module(mod_stream_management_stale_h).
+-behaviour(gen_server).
+
+-include("mongoose.hrl").
+-include_lib("stdlib/include/ms_transform.hrl").
+
+-record(state,
+        {gc_repeat_after :: non_neg_integer(),
+         gc_geriatric :: non_neg_integer()
+        }).
+
+-record(stream_mgmt_stale_h,
+        {smid :: mod_stream_management:smid(),
+         h :: non_neg_integer(),
+         stamp :: non_neg_integer()
+        }).
+
+-export([read_stale_h/1,
+         write_stale_h/2,
+         delete_stale_h/1
+        ]).
+
+-export([maybe_start/2,
+         stop/0
+        ]).
+
+%% Internal exports
+-export([start_link/2]).
+%% gen_server callbacks
+-export([init/1,
+         handle_call/3,
+         handle_cast/2,
+         handle_info/2
+        ]).
+
+-spec read_stale_h(SMID :: mod_stream_management:smid()) ->
+    {stale_h, non_neg_integer()} | {error, smid_not_found}.
+read_stale_h(SMID) ->
+    try
+        case mnesia:dirty_read(stream_mgmt_stale_h, SMID) of
+            [#stream_mgmt_stale_h{h = H}] -> {stale_h, H};
+            [] -> {error, smid_not_found}
+        end
+    catch exit:_Reason ->
+              {error, smid_not_found}
+    end.
+
+-spec write_stale_h(SMID :: mod_stream_management:smid(), H :: non_neg_integer()) ->
+    ok | {error, any()}.
+write_stale_h(SMID, H) ->
+    try
+        Stamp = timestamp_to_seconds(erlang:timestamp()),
```

Comment:
you can ask erlang directly for seconds, without asking for more precise timestamp and casting.

You can use new timestamp API:
```erlang
erlang:system_time(seconds).
```

But we use compat module still:
```erlang
p1_time_compat:system_time(seconds).
```

Reply:
There is one important thing we need to remember when using `erlang:system_time`. It's not monotonically increasing. For instance it can go back or forward in time when the system clock changes (sync or daylight saving changes). It may result in some records removed too quickly or staying in the database for too long. In this particular case, I'm not sure this causes much trouble , though.

Regarding the `p1_time_compat:system_time`. I think we no longer need it. It was need when we were in transition period between Erlang/OTP 17, 18 and 19.

Reply:
Actually, there's `erlang:monotonic_time(seconds)` [function](http://erlang.org/doc/man/erlang.html#monotonic_time-1). It perfect: it is monotonic, from the erlang standard library, returns specifically seconds; seems like just what we wanted 😃